src/pages/studyView/virtualStudy/VirtualStudy.tsx

```diff
@@ -231,6 +222,34 @@ export default class VirtualStudy extends React.Component<
         );
     }
 
+    getDefuaultDescriptionByType(dynamic: boolean) {
+        return getVirtualStudyDescription(
+            this.props.description,
+            this.props.studyWithSamples,
+            this.props.filter,
+            this.attributeNamesSet,
+            this.props.molecularProfileNameSet,
+            this.props.caseListNameSet,
+            this.props.user,
+            dynamic
+        );
+    }
+
+    @computed get description() {
+        const noCustomDescriptionProvided =
+            this.customDescription == undefined ||
+            this.customDescription ===
+                this.getDefuaultDescriptionByType(!this.dynamic);
+        if (noCustomDescriptionProvided) {
+            return this.getDefuaultDescriptionByType(this.dynamic);
+        }
+        return this.customDescription || '';
+    }
+
+    set description(description: string) {
```

Comment:
you shouldn't have a setter.  the description get above can check the prop if there is one.  i actually don't see where this called

Reply:
See `onChange` handler for the descriptions field:
```
onChange={event =>
                                                    (this.description =
                                                        event.currentTarget.value)
                                                }
```
yea, the setter is not necessary. We can use the `customDescription` like the following and get rid from the setter:
```
onChange={event =>
                                                    (this.customDescription =
                                                        event.currentTarget.value)
                                                }
```

Is that what you meant?

Reply:
yes. that would be cleaner.   i just don't like there be a setter on a property which is computed because it can create confusion.